internal/attachment/usecase/attachment_usecase.go

```diff
@@ -0,0 +1,76 @@
+package usecase
+
+import (
+	"context"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/meroedu/meroedu/internal/domain"
+	"github.com/meroedu/meroedu/pkg/log"
+)
+
+// AttachmentUseCase ...
+type AttachmentUseCase struct {
+	attachmentStore domain.AttachmentStorage
+	attachmentRepo  domain.AttachmentRepository
+	contextTimeOut  time.Duration
+}
+
+// NewAttachmentUseCase ...
+func NewAttachmentUseCase(a domain.AttachmentRepository, store domain.AttachmentStorage, timeout time.Duration) domain.AttachmentUserCase {
+	return &AttachmentUseCase{
+		attachmentRepo:  a,
+		attachmentStore: store,
+		contextTimeOut:  timeout,
+	}
+}
+
+// CreateAttachment ...
+func (usecase *AttachmentUseCase) CreateAttachment(ctx context.Context, attachment domain.Attachment) (*domain.Attachment, error) {
+	ctx, cancel := context.WithTimeout(ctx, usecase.contextTimeOut)
+	defer cancel()
+	filename := getFileName(attachment.Type)
+	if filename == nil {
+		return nil, domain.ErrUnsupportedFileType
+	}
+	attachment.Name = *filename
+	err := usecase.attachmentStore.CreateAttachment(ctx, attachment)
+	if err != nil {
+		log.Errorf("error received from usecase storage %v", err)
+		return nil, err
+	}
+	err = usecase.attachmentRepo.CreateAttachment(ctx, attachment)
+	if err != nil {
+		log.Errorf("error received from usecase repository %v", err)
+		return nil, err
+	}
+	return &attachment, nil
+}
+func getUUID() string {
+	id := uuid.New()
+	return id.String()
+}
```

Comment:
I think we should not introduce a new function to generate uuid and return its value string. Just keep it simple by using something like that"
```go
filename = uuid.New().String()
```
That makes maintainers can easier to understand code logic.

Reply:
yeah, that makes sense. Will do that.